crates/pgmq/tests/integration_test.rs

```diff
@@ -179,6 +179,38 @@ async fn test_fifo() {
     assert_eq!(read3.msg_id, 3);
 }
 
+#[tokio::test]
+async fn test_read_batch() {
+    let test_queue = "test_read_batch".to_owned();
+
+    let queue = init_queue(&test_queue).await;
+
+    // PUBLISH THREE MESSAGES
+    let msg = serde_json::json!({
+        "foo": "bar1"
+    });
+    let msg_id1 = queue.send(&test_queue, &msg).await.unwrap();
+    assert_eq!(msg_id1, 1);
+    let msg_id2 = queue.send(&test_queue, &msg).await.unwrap();
+    assert_eq!(msg_id2, 2);
+    let msg_id3 = queue.send(&test_queue, &msg).await.unwrap();
+    assert_eq!(msg_id3, 3);
+
+    let vt: i32 = 1;
+    let num_msgs = 3;
+
+    let batch = queue
+        .read_batch::<Value>(&test_queue, Some(&vt), &num_msgs)
+        .await
+        .unwrap()
+        .unwrap();
+
+    for (i, message) in batch.iter().enumerate() {
+        let index = i + 1;
+        assert_eq!(message.msg_id.to_string(), index.to_string());
+    }
```

Comment:
can you add a test that validates the visibility timeout is getting set appropriately? something like:

```rust
    let msg = queue.read::<Value>(&test_queue, Some(&vt)).await.unwrap();
    assert!(msg.is_none());
    let num_rows = rowcount(&test_queue, &queue.connection).await;
    // no messages on read but still 3 records on the table
    assert_eq!(num_rows, 3);
    ```